packages/xai/src/responses/xai-responses-language-model.ts

```diff
@@ -0,0 +1,598 @@
+      tools: xaiTools,
+      toolChoice: xaiToolChoice,
+      toolWarnings,
+    } = await prepareResponsesTools({
+      tools,
+      toolChoice,
+    });
+    warnings.push(...toolWarnings);
+
+    const baseArgs: Record<string, unknown> = {
+      model: this.modelId,
+      input,
+      max_tokens: maxOutputTokens,
+      temperature,
+      top_p: topP,
+      seed,
+      reasoning_effort: options.reasoningEffort,
+    };
+
+    if (xaiTools && xaiTools.length > 0) {
+      baseArgs.tools = xaiTools;
+    }
+
+    if (xaiToolChoice != null) {
+      baseArgs.tool_choice = xaiToolChoice;
+    }
+
+    return {
+      args: baseArgs,
+      warnings,
+      webSearchToolName,
+      xSearchToolName,
+      codeExecutionToolName,
+    };
+  }
+
+  async doGenerate(
+    options: Parameters<LanguageModelV3['doGenerate']>[0],
+  ): Promise<Awaited<ReturnType<LanguageModelV3['doGenerate']>>> {
+    const {
+      args: body,
+      warnings,
+      webSearchToolName,
+      xSearchToolName,
+      codeExecutionToolName,
+    } = await this.getArgs(options);
+
+    const {
+      responseHeaders,
+      value: response,
+      rawValue: rawResponse,
+    } = await postJsonToApi({
+      url: `${this.config.baseURL ?? 'https://api.x.ai/v1'}/responses`,
+      headers: combineHeaders(this.config.headers(), options.headers),
+      body,
+      failedResponseHandler: xaiFailedResponseHandler,
+      successfulResponseHandler: createJsonResponseHandler(
+        xaiResponsesResponseSchema,
+      ),
+      abortSignal: options.abortSignal,
+      fetch: this.config.fetch,
+    });
+
+    const content: Array<LanguageModelV3Content> = [];
+
+    const webSearchSubTools = [
+      'web_search',
+      'web_search_with_snippets',
+      'browse_page',
+    ];
+    const xSearchSubTools = [
+      'x_user_search',
+      'x_keyword_search',
+      'x_semantic_search',
+      'x_thread_fetch',
+    ];
+
+    for (const part of response.output) {
+      if (
+        part.type === 'web_search_call' ||
+        part.type === 'x_search_call' ||
+        part.type === 'code_interpreter_call' ||
+        part.type === 'code_execution_call' ||
+        part.type === 'view_image_call' ||
+        part.type === 'view_x_video_call'
+      ) {
+        let toolName = part.name;
+        if (webSearchSubTools.includes(part.name)) {
+          toolName = webSearchToolName ?? 'web_search';
+        } else if (xSearchSubTools.includes(part.name)) {
+          toolName = xSearchToolName ?? 'x_search';
+        } else if (part.name === 'code_execution') {
+          toolName = codeExecutionToolName ?? 'code_execution';
+        }
+
+        content.push({
+          type: 'tool-call',
+          toolCallId: part.id,
+          toolName,
+          input: part.arguments,
+          providerExecuted: true,
+        });
+
+        content.push({
+          type: 'tool-result',
+          toolCallId: part.id,
+          toolName,
+          result: undefined,
```

Comment:
I wonder if this could cause the error? Makes no sense to me that we enqueue a result content item, but set no actual result